util/span: use bazel for codegen, regenerate

ajwerner opened the pull request:
In #110516, the code started used a generated btree. Unfortunately, the code did not adopt bazel code generation, so missed a regeneration when the changes in #116663 merged.

Also, it seems, the change rotted such that regenerating didn't work because the entry was renamed from `frontierEntry` to `btreeFrontierEntry` without updating the go:generate comment. I imagine somebody did a manual rename.

This fixes the problem by adopting bazel generation.

Epic: None

Release note: None

ajwerner committed:
util/span: use bazel for codegen, regenerate

In #110516, the code started used a generated btree. Unfortunately, the
code did not adopt bazel code generation, so missed a regeneration when
the changes in #116663 merged.

Also, it seems, the change rotted such that regenerating didn't work
because the entry was renamed from `frontierEntry` to
`btreeFrontierEntry` without updating the go:generate comment. I imagine
somebody did a manual rename.

This fixes the problem by adopting bazel generation.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/118490)
<!-- Reviewable:end -->

ajwerner committed:
util/span: add a doc.go file

For whatever reason, the linter is unhappy that this package does
not have a comment. I don't know how we avoided this failure in
the past, but maybe it's being triggered by the fact that I'm changing
the package?

Epic: none

Release note: None

ajwerner commented:
@miretskiy can you stamp this?

miretskiy commented:
Bors r+
